### PR TITLE
fixed issues of opportunities

### PR DESCRIPTION
**Pull request by Cxristian**

### CODE REVIEW

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java [hunk 1]
```diff
@@ -163,10 +160,47 @@ public void SearchesForAnExistingOpportunity() {
...
+        JSONObject inputBody = new JSONObject();
+        String contactId = context.getContact().getId();
+        String opportunityId = context.getOpportunity().getId();
+        inputBody.put("OpportunityId", opportunityId);
+        inputBody.put("ContactId", contactId);
+        opportunityApi.setContent(inputBody);
+    }
+
+    /**
+     *
```
Comment (TheYisusGithub) [inline src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java:197]: Please comment that.

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java [hunk 2]
```diff
@@ -163,10 +160,47 @@ public void SearchesForAnExistingOpportunity() {
...
+     * @param method
```
Comment (TheYisusGithub) [inline src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java]: Please comment method.

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java (hunk 1 above)
Comment (Cxristian) [reply src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java:197]: sorry ok...

File: src/main/java/com/foundation/salesforce/entities/OpportunityContactRole.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    /**
+     * Sets the id of OpportunityContactRole.
+     *
+     * @param id for the OpportunityContactRole.
+     */
+    public void setId(String id) {
+        this.id = id;
+    }
+}
```
Comment (titor-oopart) [inline src/main/java/com/foundation/salesforce/entities/OpportunityContactRole.java]: add empty line

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java (hunk 1 above)
Comment (Cxristian) [reply src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java:197]: done

File: src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java (hunk 2 above)
Comment (Cxristian) [reply src/test/java/com/foundation/salesforce/steps/OpportunitySteps.java]: done

File: src/test/resources/features/opportunity/CreateOpportunityContactRole.feature [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+Feature: Create Opportunity Contact Role in Salesforce
+
+  @CreateOpportunity @DeleteOpportunity @DeleteContact @CreateContact @Acceptance
+  Scenario: Create a new opportunity contact role with the required data.
+    Given User set up the data with contact id
+    When User send request POST to opportunity contact role endpoint
+    Then the status code is 201
+      And the response includes the following
+        | success | true |
+    And the response passes opportunity create schema validation
```
Comment (titor-oopart) [inline src/test/resources/features/opportunity/CreateOpportunityContactRole.feature]: add empty line

File: src/test/resources/features/opportunity/FindOpportunity.feature [hunk 5]
```diff
@@ -9,3 +9,16 @@ Feature: Find Opportunity in Salesforce
...
+    Then the status code is 404
+      And the response contains the following
+        | errorCode | NOT_FOUND |
+
+  @Negative
+  Scenario: A user searches for a non-existing opportunity
+    When User searches for opportunity 0054P000007ts7CQAQ5
+    Then the status code is 404
+    And the response contains the following
+      | errorCode | NOT_FOUND |
```
Comment (alszla) [inline src/test/resources/features/opportunity/FindOpportunity.feature]: Please add a new blank line.

**Review by titor-oopart — APPROVED** [review]

File: src/main/java/com/foundation/salesforce/entities/OpportunityContactRole.java (hunk 3 above)
Comment (Cxristian) [reply src/main/java/com/foundation/salesforce/entities/OpportunityContactRole.java]: done

File: src/test/resources/features/opportunity/FindOpportunity.feature (hunk 5 above)
Comment (Cxristian) [reply src/test/resources/features/opportunity/FindOpportunity.feature]: done

File: src/test/resources/features/opportunity/CreateOpportunityContactRole.feature (hunk 4 above)
Comment (Cxristian) [reply src/test/resources/features/opportunity/CreateOpportunityContactRole.feature]: done

**Review by alszla — APPROVED** [review]

**Pull request merged**